internal/policy/policy.go

```diff
@@ -28,36 +30,72 @@ import (
 	cosignSig "github.com/sigstore/cosign/pkg/signature"
 	sigstoreSig "github.com/sigstore/sigstore/pkg/signature"
 	log "github.com/sirupsen/logrus"
+
+	"github.com/hacbs-contract/ec-cli/internal/kubernetes"
 )
 
-// WithRekorUrl returns a new instance of EnterpriseContractPolicy if an update
-// to its rekorUrl is needed. Otherwise, the same instance is returned.
-func WithRekorUrl(policy *ecc.EnterpriseContractPolicy, rekorUrl string) *ecc.EnterpriseContractPolicy {
-	if rekorUrl != "" && rekorUrl != policy.Spec.RekorUrl {
-		newPolicy := policy.DeepCopy()
-		newPolicy.Spec.RekorUrl = rekorUrl
-		log.Debugf("Updated rekor URL in policy to %q", rekorUrl)
-		return newPolicy
+var kubernetesClientCreator = kubernetes.NewClient
+
+// NewEnterpriseContractPolicy construct and return a new instance of EnterpriseContractPolicySpec.
+//
+// The policyRef parameter is expected to be either a JSON-encoded instance of
+// EnterpriseContractPolicySpec, or reference to the location of the EnterpriseContractPolicy
+// resource in Kubernetes using the format: [namespace/]name
+//
+// If policyRef is blank, an empty EnterpriseContractPolicySpec is used.
+//
+// rekorUrl and publicKey provide a mechanism to overwrite the attributes, of same name, in the
+// EnterpriseContractPolicySpec.
+func NewPolicy(ctx context.Context, policyRef, rekorUrl, publicKey string) (*ecc.EnterpriseContractPolicySpec, error) {
+	var policy *ecc.EnterpriseContractPolicySpec
+
+	if policyRef == "" {
+		log.Debug("Using an empty EnterpriseContractPolicy")
+		// Default to an empty policy instead of returning an error because the required
+		// values, e.g. PublicKey, may be provided via other means, e.g. publicKey param.
+		policy = &ecc.EnterpriseContractPolicySpec{}
+	} else if strings.Contains(policyRef, "{") {
```

Comment:
Nitpick: trim leading whitespace in `policyRef`?

Reply:
This is using `Contains`, not `HasPrefix`, so it should work regardless. I picked the `{` character because it's likely the first one to be found (and it's also not a valid character for a resource name).

Reply:
Yeah you're right, I read that as `strings.HasPrefix`. `strings.Contains` should be good enough